packages/@aws-cdk/aws-cognito/README.md

```diff
@@ -53,6 +66,265 @@ This module is part of the [AWS Cloud Development Kit](https://github.com/aws/aw
+  authenticatedRole: Role.fromRoleArn('arn:aws:iam::123456789012:role/my-authenticated-role'),
+  unauthenticatedRole: Role.fromRoleArn('arn:aws:iam::123456789012:role/my-unauthenticated-role'),
+});
+```
+
+
+### Federated Identity Providers
+
+You can associate identity providers with an Identity Pool by first associating them with a Cognito User Pool or by associating the provider directly with the identity pool.
+
+#### Associating a Provider Through a User Pool
+
+```ts
+const userpool = new cognito.UserPool(this, 'Pool');
+
+const provider = new cognito.UserPoolIdentityProviderAmazon(this, 'Amazon', {
+  clientId: 'amzn-client-id',
+  clientSecret: 'amzn-client-secret',
+  userPool: userpool,
+});
+
+new cognito.IdentityPool(this, 'myidentitypool', {
+  identityPoolName: 'myidentitypool',
+  authenticatedRole: Role.fromRoleArn('arn:aws:iam::123456789012:role/my-authenticated-role'),
+  unauthenticatedRole: Role.fromRoleArn('arn:aws:iam::123456789012:role/my-unauthenticated-role'),
+  userPools: [userpool]
+});
+```
+
+When associating a user pool with an identity pool, a `UserPoolClient` is automatically created to handle the user pool's providers. If you want to control the the settings of the `UserPoolClient`, you can do that using `IdentityPoolProps.defaultClientOptions`:
+
+```ts
+new cognito.IdentityPool(this, 'myidentitypool', {
+  identityPoolName: 'myidentitypool',
+  authenticatedRole: Role.fromRoleArn('arn:aws:iam::123456789012:role/my-authenticated-role'),
+  unauthenticatedRole: Role.fromRoleArn('arn:aws:iam::123456789012:role/my-unauthenticated-role'),
+  userPools: [userpool],
+  defaultClientOptions: {
+    userPoolClientName: 'my-user-pool-client',
+    disableOath: true,
+    refreshTokenValidity: Duration.days(14),
+  },
+});
+```
+
+You can also associate user pools with an identity pool after instantiation. The Identity Pool's `addUserPool` method returns the User Pool Client that has been created:
+
+```ts
+const userPoolClient = identityPool.addUserPool(userpool, {
+  userPoolClientName: 'my-user-pool-client',
+  disableOath: true,
+  refreshTokenValidity: Duration.days(14),
+});
+```
+
+#### Associating a Provider Directly
+
+You can associate with one or more [external identity providers](https://docs.aws.amazon.com/cognito/latest/developerguide/external-identity-providers.html) directly with an identity pool using `IdentityPoolProps.supportedLoginProviders`:
+
+```ts
+new cognito.IdentityPool(this, 'myidentitypool', {
+  identityPoolName: 'myidentitypool',
+  authenticatedRole: Role.fromRoleArn('arn:aws:iam::123456789012:role/my-authenticated-role'),
+  unauthenticatedRole: Role.fromRoleArn('arn:aws:iam::123456789012:role/my-unauthenticated-role'),
+  supportedLoginProviders: {
+    amazon: 'amzn1.application.12312k3j234j13rjiwuenf',
+    facebook: '1234567890123',
+    google: '12345678012.apps.googleusercontent.com',
+    apple: 'com.yourappleapp.auth',
+    twitter: 'my-twitter-id;my-twitter-secret',
+  },
```

Comment:
This feels awkward. Can we do something like this instead -

```ts
supportedLoginProviders: {
  amazon: { appId: 'xxx' },
  twitter: { consumerKey: 'xxx', consumerSecret: 'yyy  },
  ...
}
```

Reply:
The word `supported` is redundant. Just `loginProviders` should be sufficient.

The AWS console calls this 'Authentication Providers'. Perhaps we should use that, and fold in user pool, OpenId, SAML and custom into this block.

Reply:
I like the idea of calling them `authenticationProviders` and including `customProviders` in there, my hesitation with user pools, Open Id and Saml were that they are all arrays, suggesting that you could attach more than one. This does not appear to be possible in the console with Open Id and Saml, but multiple user pools can be added there. It would be cleaner to include them all in that block though, so I'll make that change as well.